packages/ERTP/src/issuer.chainmail

```diff
@@ -0,0 +1,209 @@
+ * assets all have the same kind, which is called a Brand.
+ */
+interface Mint (Amount (Extent)) {
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /**
+   * Create a new Payment containing newly minted amount. 
+   */
+  mintPayment(newAmount :Amount) -> (Payment);
+}
+
+/**
+ * Purses hold amount of digital assets of the same brand, but unlike Payments, they are
+ * not meant to be sent to others. To transfer digital assets, a
+ * Payment should be withdrawn from a Purse. A purse's balance can rise
+ * and fall, through the action of deposit() and withdraw().
+ *
+ * The primary use for Purses and Payments is for currency-like and goods-like
+ * digital assets, but they can also be used to represent other kinds of rights, such
+ * as the right to participate in a particular contract.
+ */
+interface Purse (Amount) {
+
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /** Get the amount contained in this purse, confirmed by the issuer. */
+  getBalance() -> (Amount);
+
+  /**
+   * Deposit all the contents of srcPayment into this purse, returning the
+   * amount. If the optional argument amount does not equal the balance of srcPayment,
+   * throw error.
+   */
+  deposit(srcPayment :Payment, amount :Amount) -> (Amount);
+
+  /** Withdraw amount from this purse into a new Payment. */
+  withdraw(amount :Amount) -> (Payment);
+
+}
+
+/**
+ * Payments hold amount of digital assets of the same brand in transit. Payments can
+ * be deposited in purses, split into multiple payments, combined, and
+ * claimed (getting an exclusive payment). Payments are linear, meaning
+ * that either a payment has its full original balance, or it is used up
+ * entirely. It is impossible to partially use a payment. 
+ *
+ * Payments are often received from other actors and therefore should
+ * not be trusted themselves. To get the balance of a payment, use the
+ * trusted issuer: issuer.getBalance(payment),
+ *
+ * Payments can be converted to Purses by getting a trusted issuer and
+ * calling `issuer.makeEmptyPurse()` to create a purse, then `purse.deposit(payment)`.
+ */
+interface Payment (Amount) {
+
+  /** 
+   * Get the allegedBrand, indicating the kind of digital asset this
+   * payment purports to be, and which issuer to use. Because payments 
+   * are not trusted, any method calls on payments should be treated
+   * with suspicion and verified elsewhere.
+   */ 
+  allegedBrand() -> (Brand);
+}
+
+/** 
+ * All of the difference in how digital asset amount are manipulated can be reduced to
+ * the behavior of the math on extents. We extract this
+ * custom logic into mathHelpers. MathHelpers are about extent
+ * arithmetic, whereas AmountMath is about amounts, which are the
+ * extents labeled with a brand. AmountMath use mathHelpers to do their extent arithmetic,
+ * and then brand the results, making a new amount. 
+ */ 
+interface mathHelpers () {
```

Comment:
This file should explain `AmountMath` and not explain `mathHelpers`.

Reply:
but there's another file for AmountMath. Do you mean we should remove this? Maybe put it in another file?

Reply:
No, sorry, I missed that. Nevermind. Given that, I don't actually mind mathHelpers being here.

Resolving.